hermes/handlers/api.py

```diff
@@ -857,26 +880,82 @@ def post(self):
             self.write_error(400, message="Bad event type")
             return
 
-        host = Host.get_host(self.session, hostname)
+        # If a host query was specified, we need to talk to the external
+        # query server to resolve this into a list of hostnames
+        if "hostQuery" in self.jbody:
+            query = self.jbody["hostQuery"]
+            response = PluginHelper.request_get(params={"query": query})
+            if response.json()["status"] == "ok":
+                hostnames.extend(response.json()["results"])
+
+        # We need to create a list of hostnames that don't have a Host record
+        new_hosts_needed = list(hostnames)
+        hosts = (
+            self.session.query(Host).filter(Host.hostname.in_(hostnames)).all()
+        )
+        for host in hosts:
+            new_hosts_needed.remove(str(host.hostname))
+
+        # if we need to create hosts, do them all at once
+        if new_hosts_needed:
+            Host.create_many(self.session, new_hosts_needed)
+            hosts = (
+                self.session.query(Host).filter(
+                    Host.hostname.in_(hostnames)
+                ).all()
+            )
 
-        if host is None:
-            host = Host.create(self.session, hostname)
+        if len(hosts) == 0:
```

Comment:
This is usually written as `if not hosts` which would match both the `None` case and the `[]` case.